From 03881dcf31b358675498a463c98ce14caf6d21f6 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Sun, 24 Jan 2021 14:02:20 +0530
Subject: [PATCH 3/7] sm8350-common: Switch to dot product CPU variant

 * lisa:/ $ cat /proc/cpuinfo | grep -m 1 Features
   Features: fp asimd evtstrm aes pmull sha1 sha2 crc32
   atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp

 * From this, it's evident that our CPU (Snapdragon 888)
   supports all the features of ARMv8.2-DotProd [1].
   Moreover AOSP sets this arch variant in their generic
   cortex-a55 ART target as well [2]. This should unlock
   some optimizations in different code paths.

[1]:
https://en.wikichip.org/wiki/arm/armv8#ARMv8_Extensions_and_Processor_Features
[2]:
https://android.googlesource.com/device/generic/art/+/refs/heads/master/armv8_cortex_a55/BoardConfig.mk#23

Change-Id: I484bb2bcc181c62e29c04bc6e467e5edea8ac0d8
---
 BoardConfigCommon.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 297ce06a..0134c70a 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -22,7 +22,7 @@ AB_OTA_PARTITIONS += \
 
 # Architecture
 TARGET_ARCH := arm64
-TARGET_ARCH_VARIANT := armv8-2a
+TARGET_ARCH_VARIANT := armv8-2a-dotprod
 TARGET_CPU_ABI := arm64-v8a
 TARGET_CPU_ABI2 :=
 TARGET_CPU_VARIANT := cortex-a76
-- 
2.49.0

